agenta-backend/agenta_backend/services/db_manager.py

```diff
@@ -296,13 +296,41 @@ async def list_apps(**kwargs: dict) -> List[App]:
     if user is None:
         return []
 
-    query_expression = query.eq(AppVariantDB.user_id, user.id) & query.eq(
-        AppVariantDB.is_deleted, False
-    )
-    apps: List[AppVariantDB] = await engine.find(AppVariantDB, query_expression)
-    apps_names = [app.app_name for app in apps]
-    sorted_names = sorted(set(apps_names))
-    return [App(app_name=app_name) for app_name in sorted_names]
+    # Get the organizations the user belongs to
+    user_organizations = await get_user_organizations(user)
+
+    if user_organizations is not None:
+        # Get AppVariantDB documents where user_id matches the specified user's id and is not deleted
+        user_apps = await engine.find(
```

Comment:
isn't this the same query expression from line 327? why are we repeating ourselves.

Isnt't the logic

user_apps = 
if orga_apps
   orga_apps =
return user_apps+orga_apps

Reply:
but why the user wouldn't have an organization unless we have a bug?